pymeasure/instruments/temptronic/temptronic_eco560.py

```diff
@@ -0,0 +1,86 @@
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+"""
+Implementation of an interface class for ThermoStream® Systems devices.
+Reference Document for implementation:
+ECO-560/660 ThermoStream® Systems
+Operators Manual
+Revision B
+September, 2018
+"""
+
+from pymeasure.instruments.temptronic.temptronic_base import ATSBase
+from pymeasure.instruments.instrument import Instrument
+from pymeasure.instruments.validators import truncated_range
+
+#
+# NO_DUT_SENSOR_SELECTED = 16384      # bit 14 – no DUT sensor selected
+# IMPROPER_SOFTWARE_VERSION = 8192    # bit 13 – reserved
+# BVRAM_FAULT = 4096                  # bit 12 – BVRAM fault
+# NVRAM_FAULT = 2048                  # bit 11 – NVRAM fault
+# PURGE_HEAT_FAILURE = 1024           # bit 10 – purge heat failure
+# FLOW_SENSOR_HARDWARE_ERROR = 512    # bit 9  – flow sensor hardware error
+# DUT_OPEN_LOOP = 256                 # bit 8  – dut open loop
+# INTERNAL_ERROR = 128                # bit 7  – internal error
+# OPEN_PURGE_TEMPERATURE_SENSOR = 64  # bit 6  – open purge temperature sensor
+# NO_PURGE_FLOW = 32                  # bit 5  – no purge flow
+# LOW_INPUT_AIR_PRESSURE = 16         # bit 4  – low input air pressure
+# LOW_FLOW = 8                        # bit 3  – low flow
+# SETPOINT_OUT_OF_RANGE = 4           # bit 2  – setpoint out of range
+# AIR_OPEN_LOOP = 2                   # bit 1  – air open loop
+# OVERHEAT = 1                        # bit 0  – overheat
+# OK = 0                              # ok state
+#
+
+
+class ECO560(ATSBase):
+    """Represent the TemptronicECO560 instruments.
+    """
+
+    temperature_limit_air_low = Instrument.control(
+        "LLIM?", "LLIM %g",
+        """Control lower air temperature limit.
+
+        :type: float
+
+        Valid range between -150 to 25 (°C). Setpoints below current value cause
+        “out of range” error in TS.
+        """,
+        validator=truncated_range,
+        values=[-150, 25]
+    )
```

Comment:
Afaict, you are redefining this property only to change the `values` entry. Luckily, nowadays pymeasure can use "dynamic properties" to do this without duplicating code! :-) See [this example](https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments.html#family-of-instruments-with-similar-features) -- you have to turn the base class' `temperature_limit_air_low` into a dynamic property by adding `dynamic=True`, then redefine the class attribute `temperature_limit_air_low_values` in here.

Reply:
I've implemented this change across all temptronic instruments, which were also replicating this property. I am a little confused by how the structure works though, how does the addition of '_values' refer back to the original property definition?

Reply:
Worked it out

Reply:
> how does the addition of '_values' refer back to the original property definition?

~~Magic~~ Python being a very dynamic language :sweat_smile:

Reply:
> I've implemented this change across all temptronic instruments, which were also replicating this property. 

Great call on also checking the other instruments for this! :clap: